File: api/application/src/main/java/net/corda/v5/application/marshalling/JsonMarshallingService.java
package net.corda.v5.application.marshalling;

import net.corda.v5.application.flows.ClientStartableFlow;
import net.corda.v5.base.annotations.DoNotImplement;

/**
 * {@link JsonMarshallingService} marshalls to and from JSON using the registered JSON mapper.
 * <p>
 * The platform will provide an instance of {@link JsonMarshallingService} to flows via property injection.
 * <p>
 * Example usage:
 * @see ClientStartableFlow
 */
@DoNotImplement
public interface JsonMarshallingService extends MarshallingService {
}

File: application/src/main/kotlin/net/corda/v5/application/marshalling/JsonMarshallingService.kt
package net.corda.v5.application.marshalling

import net.corda.v5.application.flows.ClientStartableFlow
import net.corda.v5.base.annotations.DoNotImplement

/**
 * [JsonMarshallingService] marshalls to and from JSON using the registered JSON mapper.
 *
 * The platform will provide an instance of [JsonMarshallingService] to flows via property injection.
 *
 * Example usage:
 * @see ClientStartableFlow
 */
@DoNotImplement
interface JsonMarshallingService : MarshallingService
